### PR TITLE
update only packages with changes

### PR DESCRIPTION
**Pull request by maximmi**
## what
* update only packages with changes
* update all packages if `apk`, `tasks` or `codefresh` folders updated
* upgrade of variant

## why
* avoid rebuilding all packages if only some was updated

### CODE REVIEW

**Comment by osterman** [comment]
Fix merge conflict

**Comment by osterman** [comment]
rename `codefresh.yaml` to something more descriptive like `docker.yaml` or what it does

File: vendor/Makefile [hunk 1]
```diff
@@ -6,16 +6,14 @@ BUILD_LIST_TARGET ?= list/all
...
 list/updated:
-	@git --no-pager diff --name-only $(CURRENT_BRANCH) `git merge-base $(CURRENT_BRANCH) origin/$(TARGET_BRANCH)` | \
-		xargs -n 1 -r dirname | \
-		sort -u | \
-		grep -E '^vendor/' | \
-		sed 's,vendor/,,' | \
-		cut -d/ -f1 | \
-		sort -u | \
-		grep -E '.+'
+	$(eval CHANGES=$(shell git --no-pager diff --name-only origin/$(TARGET_BRANCH) | \
```
Comment (osterman) [inline vendor/Makefile]: Add `#` comments that describe in plain english what this does

Comment (osterman) [reply vendor/Makefile]: To workflows:

1) if `apk|tasks|codefresh` changes, then rebuild all
2) if `vendor` changes, then just rebuild those changed/updated packages

**Comment by Nuru** [comment]
If timestamps are not working for dependency tracking,  consider [this approach](https://www.cmcrossroads.com/article/rebuilding-when-files-checksum-changes) to track file contents via hashes.

**Comment by osterman** [comment]
Closing this in favor of a using different `Makefile` centric strategy:

1. Use a codefresh shared volume to persist state
2. create a target in each `Makefile` like 

```
$(PACKAGE_NAME)-$(PACKAGE_VERSION)-r$(PACkAGE_RELEASE).apk: apk/prepare apk/checksum apk/build apk/clean`

apk: $(PACKAGE_NAME)-$(PACKAGE_VERSION)-r$(PACkAGE_RELEASE).apk
```
If we do this, then we don't need to muck around with the greping and awking to figure out what changed. We only build packages which don't yet exist. This is all derived automatically by `make`, because make will only call  `$(PACKAGE_NAME)-$(PACKAGE_VERSION)-r$(PACkAGE_RELEASE).apk` if the file does not exist.
